tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs

```diff
@@ -201,8 +208,16 @@ private GetRcmRequest BuildRequest(IDictionary<string, Product> products)
                 configStates.Add(new RcmConfigState(cache.Path.Id, cache.Version, cache.Path.Product, cache.ApplyState, cache.Error));
             }
 
+            // capabilitiesArray needs to be big endian
+            // a first for me, until now I had never worked on a code base with an endian issue ...
+#if NETCOREAPP
+            var capabilitiesArray = _capabilities.ToByteArray(true, true);
+#else
+            var capabilitiesArray = _capabilities.ToByteArray().Reverse().ToArray();
```

Comment:
This is pretty expensive, I'd be inclined to do the reverse manually. 

Also, if this is only reason to use `BigInteger`, I'd be _inclined_ to just use `BinaryPrimitives` to serialize/read with endiness. Then again, assuming this works, :shipit:

Reply:
I feel the choice of json as format will mean the that this particular code path can't really be made high performance. We'll only be paging RC about every 5 seconds, so it's not on the hot path and don't think the resources consumed will significantly impact the client app.

That said, I've removed the linq calls and replaced with `Array.Reverse` to a void a number of additional allocations, for .NET Framework users.

The choice of `BigInteger` was about avoiding future errors. My plan was to replace the single instance of `BitVector32` with a list once we past 32 capabilities, however, if managed to get endianness wrong for single instance the odds of creating bugs when using a lists seems quite high.